### DISCUSSION TITLE
Status code: 400, previous solutions not working

### DISCUSSION THREAD

**Discussion by j-anderson-sci-comm** [cjbarrie/academictwitteR#230] [Q&A]
I attempting to pull tweets from a list of users with the following code:

influencer_tweets <-
  get_all_tweets(
    users = influencer_accounts,
    start_tweets = "2020-08-20T00:00:00Z",
    end_tweets = "2021-08-20T00:00:00Z",
    file = "C:/Users/josht/Desktop/sci_influencer_tweets.csv",
    data_path = "influencertweets/",
    n = 1000,
  )

and I get the following error message:

Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400
In addition: Warning message:
Directory already exists. Existing JSON files may be parsed and returned, choose a new path if this is not intended. 

I saw that the previous thread on what seemed to be the same issue (https://github.com/cjbarrie/academictwitteR/issues/183), but none of the solutions suggested there seem to work. I have checked/tried:

-the bearer token in the r environment is set to BEARER_TOKEN=MY_TOKEN
-restarting r
-downloading academictwitteR through github and cran using 0.2.1
-setting the bearer token in the call

And I am still receiving the same error. Any suggestions?

**Comment by chainsawriot** [comment]
@j-anderson-sci-comm Please let us know what are in your `influencer_accounts`.

**Reply by j-anderson-sci-comm** [reply]
This is a data frame of width 1 and length 50 that contains the usernames of 50 twitter users (c("stephen_wolfram", "Khairykj", "Elise_Andrew"...)

**Reply by j-anderson-sci-comm** [reply]
I actually just edited the code to setting the query with a vector of the accounts and it is working now, thank you so much!

**Reply by aybukeatalay** [reply]
Hi, I am also getting the same error although I've put the user names in vectors. 

get_all_tweets(
    users = c('RTErdogan', 'tcbestepe','cbabdullahgul',
              'kilicdarogluk'),
    is_retweet=FALSE,
    start_tweets = "2021-01-01T00:00:00Z",
    end_tweets = "2021-10-25T00:00:00Z", bearer_token=bearer_token,
    data_path = "C:/Users/s1878387/Desktop/influencer_tweets" )

the error I get: Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400. 

Any suggestions?

**Reply by cjbarrie** [reply]
The following code works for me:

```r

tweets <- get_all_tweets(
  users = c('RTErdogan', 'tcbestepe','cbabdullahgul',
            'kilicdarogluk'),
  is_retweet=FALSE,
  start_tweets = "2021-01-01T00:00:00Z",
  end_tweets = "2021-10-25T00:00:00Z",
  n = Inf)

```

Note you will only get 100 tweets using this code as that is the default. To get all tweets, set `n = Inf`

**Reply by aybukeatalay** [reply]
Thanks for the help and the tip!

I tried dividing my users into smaller groups and now its working. I initially requested tweets from 65 users and it did not work. Up to 49 users in one query worked, more than 49 gives the status code: 400 error.

**Reply by anastasia-menshikova** [reply]
I have the same issue with the N of users, only requests for up to 49 users do not throw the error. There is no details on it in the documentation, does it have to do with Twitter API limits? Would be great to have this clarified, thanks!

**Reply by aybukeatalay** [reply]
I also received the same error for get_liking_users function and also wondering if the error is related to Twitter API or is  there again a quota (in academictwitteR) for the number of tweet ids (as vector format) included in the query.

**Reply by shmuhammadd** [reply]
@anmenshikova  @aybukeatalay  See this https://github.com/cjbarrie/academictwitteR/issues/184#issuecomment-875760868

**Reply by larissabr** [reply]
> setting the query with a vector of the accounts

Excuse my ignorance, but what does that mean? Having the same problem and desperately looking to fix it.

**Reply by aybukeatalay** [reply]
Means that if you want to collect tweets from more than one account, you write it in a vector format which is c(a,b,c). For instance:

get_all_tweets(
users = c('user1', 'user2','user3'),
is_retweet=FALSE,
start_tweets = "2021-01-01T00:00:00Z",
end_tweets = "2021-10-25T00:00:00Z", bearer_token=bearer_token,
data_path = your data path

P.s.: don't forget to put  ' ' for each account

**Reply by larissabr** [reply]
Gotcha, thanks! I'm just looking at one account, so my problem remains.
